### ISSUE TITLE
lambda expression does not handle "." operator for dictionary

### ISSUE THREAD

**Issue by honnix** [SWI-Prolog/swipl-devel#123]
Following query will result an error
```
?- maplist([In,Out]>>atom_string(Out, In.a), [_{a:1},_{a:2}], L).
ERROR: Arguments are not sufficiently instantiated
```

while following is accepted
```
?- maplist([In,Out]>>(get_dict(a, In, V), atom_string(Out, V)), [_{a:1},_{a:2}], L).
L = ['1', '2'].
```

It looks like `library(yall)` and dictionary operator doesn't play well.

Version:
```
SWI-Prolog version 7.3.12 for x86_64-darwin15.0.0
```

**Comment by JanWielemaker** [comment]
Agree.  Pile of work is a bit too high to look into that on short term.

**Comment by honnix** [comment]
@JanWielemaker Thanks and t's not urgent at all.

**Comment by JanWielemaker** [comment]
This issue has been mentioned on **SWI-Prolog**. There might be relevant details there:

https://swi-prolog.discourse.group/t/maplist-using-a-lambda-on-a-lists-of-dicts-doesnt-work-as-expected/9007/3

**Comment by JanWielemaker** [comment]
This is a fundamental limitation of how Prolog meta predicates and code-is-data works.    Can only be fixed with additional syntax that tell us when the evaluation needs to take place.     A helper predicate can be used for disambiguation.
